pdata/internal/generated_immutable_slice.go

```diff
@@ -0,0 +1,162 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "model/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "go run model/internal/cmd/pdatagen/main.go".
+
+package internal
+
+// ImmutableByteSlice represents a []byte slice that cannot be mutated.
+type ImmutableByteSlice struct {
+	value []byte
+}
+
+// NewImmutableByteSliceMoveFrom creates a new ImmutableByteSlice by replacing an underlying value of the []byte slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableByteSliceCopyFrom if it needs to be preserved.
+func NewImmutableByteSliceMoveFrom(val *[]byte) ImmutableByteSlice {
+	is := ImmutableByteSlice{}
+	if val != nil {
+		is.value = *val
+		*val = nil
+	}
+	return is
+}
+
+// NewImmutableByteSliceCopyFrom creates a new ImmutableByteSlice by copying the provided []byte slice.
+// Use NewImmutableByteSliceMoveFrom if the provided slice is not needed anymore.
+func NewImmutableByteSliceCopyFrom(val []byte) ImmutableByteSlice {
+	is := ImmutableByteSlice{}
+	if len(val) != 0 {
+		is.value = make([]byte, len(val))
+		copy(is.value, val)
+	}
+	return is
+}
+
+// RawSlice returns a copy of the []byte slice.
+func (is ImmutableByteSlice) RawSlice() []byte {
+	if len(is.value) == 0 {
+		return nil
+	}
+	val := make([]byte, len(is.value))
+	copy(val, is.value)
+	return val
+}
+
+// Len returns length of the []byte slice value.
+func (is ImmutableByteSlice) Len() int {
+	return len(is.value)
+}
+
+// At returns an item from particular index.
+func (is ImmutableByteSlice) At(i int) byte {
+	return is.value[i]
+}
+
+// ImmutableFloat64Slice represents a []float64 slice that cannot be mutated.
+type ImmutableFloat64Slice struct {
+	value []float64
+}
+
+// NewImmutableFloat64SliceMoveFrom creates a new ImmutableFloat64Slice by replacing an underlying value of the []float64 slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableFloat64SliceCopyFrom if it needs to be preserved.
+func NewImmutableFloat64SliceMoveFrom(val *[]float64) ImmutableFloat64Slice {
+	is := ImmutableFloat64Slice{}
+	if val != nil {
+		is.value = *val
+		*val = nil
+	}
+	return is
+}
+
+// NewImmutableFloat64SliceCopyFrom creates a new ImmutableFloat64Slice by copying the provided []float64 slice.
+// Use NewImmutableFloat64SliceMoveFrom if the provided slice is not needed anymore.
+func NewImmutableFloat64SliceCopyFrom(val []float64) ImmutableFloat64Slice {
+	is := ImmutableFloat64Slice{}
+	if len(val) != 0 {
+		is.value = make([]float64, len(val))
+		copy(is.value, val)
+	}
+	return is
+}
+
+// RawSlice returns a copy of the []float64 slice.
+func (is ImmutableFloat64Slice) RawSlice() []float64 {
```

Comment:
I think we use the "AsRaw" pattern? (see the `pcommon.Value` as an example)

Reply:
Sounds good. updated